File: search_insert_position/sol.c
int mid_(int x, int y)
{
    return (x + y) / 2;
}

int searchInsert(int *nums, int numsSize, int target)
{
    int left, right;
    left = 0;
    right = numsSize - 1;
    while (left <= right)
    {
        int mid = mid_(left, right);
        if (nums[mid] == target)
        {
            return mid;
        }
        else if (nums[mid] < target)
        {
            left = mid + 1;
        }
        else
        {
            right = mid - 1;
        }
    }
    return left;

File: GS/minimum_in_rotated_array/sol.py
class Solution:
    def findMin(self, nums: list[int]) -> int:
        l, r = 0, len(nums) - 1
        while l < r:
            mid = (l + r) // 2
            if nums[mid] < nums[r]:
                r = mid
            else:
                l = mid + 1
        return nums[l]